hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/handlers/BucketHandler.java

```diff
@@ -183,8 +183,14 @@ public static BucketHandler getBucketHandler(
                 OzoneStorageContainerManager reconSCM,
                 OmBucketInfo bucketInfo) throws IOException {
 
-    return new FSOBucketHandler(reconNamespaceSummaryManager,
+    if (bucketInfo.getBucketLayout()
+            .equals(BucketLayout.FILE_SYSTEM_OPTIMIZED)) {
+      return new FSOBucketHandler(reconNamespaceSummaryManager,
               omMetadataManager, reconSCM, bucketInfo);
+    } else {
```

Comment:
have to check for legacy here

Reply:
There is no OBSBucketHandler yet. If we check for legacy, what will we call in the else statement?